### PR TITLE
Viewing party new page

### LINKED ISSUE #6: New Viewing Party Page
As an authenticated user,
When I visit the new viewing party page,
I should see a form with the following:
- [x] Movie Title (that is un-editable)
- [x] Duration of Party with a default value of movie runtime in minutes
- [x] When: field to select date
- [x] Start Time: field to select time
- [x] Checkboxes next to each friend (if user has friends)
- [x]  Button to create a party

**Details** When the party is created, the authenticated user should be redirected back to the dashboard where the new event is shown. The event should also be seen by any friends that were invited when they log in.

### PR DESCRIPTION
**Pull request by RetynaGirl**
Closes #6 

* Adds party and party guest models and migrations
* Adds page for making a new party and adding guests
* some refactoring
* All tests pass

### CODE REVIEW

File: app/services/movie_api/tmdb_interaction.rb [hunk 1]
```diff
@@ -54,10 +54,9 @@ def self.movie_by_id(id)
 
     data = conn.get("movie/#{id}")
     results = []
```
Comment (s-steel) [inline app/services/movie_api/tmdb_interaction.rb:56]: I'll delete this empty array now that it's not needed.  Nice change to that.

**Review by s-steel — APPROVED** [review]

File: app/controllers/viewing_parties_controller.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      @party = Party.create!(party_data)
+
+      params[:party][:guests].keys.each do |guest_id|
+        @party.guests << User.find(guest_id)
+      end
+    rescue ActiveRecord::RecordInvalid
+      flash.now[:error] = 'Please fill out all fields with proper information'
+      @user = current_user
+      render :new
+    end
```
Comment (jdcarey128) [inline app/controllers/viewing_parties_controller.rb:23]: I'm not familiar with this syntax. Can we go over it sometime today during our check in?

File: app/models/user.rb [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
   has_many :followed, through: :followers_ref
-  has_many :followed_ref, foreign_key: :followed_id, class_name: "Friendship"
+  has_many :followed_ref, foreign_key: :followed_id, class_name: 'Friendship', dependent: :destroy
   has_many :followers, through: :followed_ref
+  has_many :party_guests, dependent: :destroy
+  has_many :parties, through: :party_guests
 
   validates :email, uniqueness: true, presence: true
-  validates_presence_of :password_confirmation
+  validates :password_confirmation, presence: true
```
Comment (jdcarey128) [inline app/models/user.rb:10]: Isn't this refactor saying the same thing? Curious if there is a reason to change it.

**Review by jdcarey128 — APPROVED** [review]
Looks great! Nice work on this. Definitely would like to go over a couple of things that you did here in person.

**Pull request merged**